packages/flutter/lib/src/material/button.dart

```diff
@@ -451,35 +452,111 @@ class MaterialButton extends StatelessWidget {
 ///
 /// The primary purpose of this widget is to allow padding around [Material] widgets
 /// to trigger the child ink feature without increasing the size of the material.
-class _ButtonRedirectingHitDetectionWidget extends SingleChildRenderObjectWidget {
-  const _ButtonRedirectingHitDetectionWidget({
+class _ButtonPadding extends SingleChildRenderObjectWidget {
```

Comment:
It might be possible to make the description of this class a little clearer, since the name is a little misleading (maybe call it _InputPadding?) Like:

Redirect taps that occur in the padded area around the child to the center of the child. This increases the size of the button and the button's "tap target" but not its material or its ink splashes.

Reply:
Accepted doc and naming suggestion